### PR TITLE
[INLONG-8106][DataProxy] Optimize ConfigManager implementation ( part one )

### LINKED ISSUE #8106: [Improve][DataProxy] Optimize ConfigManager implementation ( part one )
### Description

Optimize the weight.properties metadata, black and white name unit data, and groupId num2Name metadata saved by ConfigManager

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gosonzhang**
- Fixes #8106 

Optimize the implementation related to ConfigManager

1. Add the replacement class PropertiesHolder of the PropertiesConfigHolder class
2. Add WeightConfigHolder class to manage weight.properties metadata
3. Add BlackListConfigHolder and WhiteListConfigHolder to manage black and white name unit data
4. Add the GroupIdNumConfigHolder class to manage the metadata of the digital mapping names of groupId and streamId
5. Modify the ConfigManager class and adjust the metadata access API for the metadata of weight.properties metadata, groupId, and streamId's digital mapping name
6. Adjust some log output printing

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java [hunk 1]
```diff
@@ -145,11 +143,9 @@ public void configure(Context context) {
                 this.slaveChannels.add(channel);
             }
         }
-        LOG.info("masters:" + this.masterChannels);
-        LOG.info("orders:" + this.orderChannels);
-        LOG.info("slaves:" + this.slaveChannels);
-        LOG.info("transfers:" + this.transferChannels);
-        LOG.info("agentFileMetrics:" + this.agentFileMetricChannels);
-        LOG.info("slaMetrics:" + this.slaMetricChannels);
+        LOG.info(
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java:146]: Good job!

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/BlackListConfigHolder.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.dataproxy.config.holder;
+
+/**
+ * save black list configure to list
+ */
+public class BlackListConfigHolder extends VisitConfigHolder {
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/BlackListConfigHolder.java:23]: WhiteList and BlackList have specific discriminatory meanings, can they be modified to AllowList or BlockList?

Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/BlackListConfigHolder.java:23]: Blacklist and whitelist are commonly used conventions

**Review by luchunliang — APPROVED** [review]
LGTM

**Review by baomingyu — APPROVED** [review]
+1

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/BlackListConfigHolder.java (hunk 2 above)
Comment (healchow) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/BlackListConfigHolder.java:23]: > Blacklist and whitelist are commonly used conventions

Yes, but now more new usages are changed to allow lists and block lists.

Similarly, the default master branch of GitHub is changed from the master branch to the main branch~

**Pull request merged**